iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractAreaTest.java

```diff
@@ -0,0 +1,294 @@
+        Network subnetwork = network.createSubnetwork("subnetwork_id", "Subnetwork", "json");
+        VoltageLevel sn1VL1 = subnetwork.newVoltageLevel()
+                .setId("sub1_vl1")
+                .setNominalV(400.0)
+                .setTopologyKind(TopologyKind.NODE_BREAKER)
+                .add();
+
+        Throwable e = assertThrows(PowsyblException.class, () -> biddingZoneA.addVoltageLevel(sn1VL1));
+        assertEquals("VoltageLevel sub1_vl1 cannot be added to Area bza. They do not belong to the same network or subnetwork", e.getMessage());
+    }
+
+    @Test
+    void addAreaBoundaryTerminal() {
+        Terminal terminal = network.getLoad("LOAD").getTerminal();
+        biddingZoneA.newAreaBoundary()
+                .setTerminal(terminal)
+                .setAc(true)
+                .add();
+        Map<Object, Boolean> expectedBoundaries = Map.of(terminal, true);
+        assertBoundaries(expectedBoundaries, biddingZoneA);
+    }
+
+    @Test
+    void addAreaBoundaryDanglingLine() {
+        VoltageLevel vlhv1 = network.getVoltageLevel("VLHV1");
+        DanglingLine danglingLine = vlhv1.newDanglingLine()
+                .setId("danglingLine")
+                .setP0(0.0)
+                .setQ0(0.0)
+                .setR(0.0)
+                .setX(0.0)
+                .setBus("NHV1")
+                .add();
+        biddingZoneA.newAreaBoundary()
+                .setBoundary(danglingLine.getBoundary())
+                .setAc(true)
+                .add();
+        Map<Object, Boolean> expectedBoundaries = Map.of(danglingLine.getBoundary(), true);
+        assertBoundaries(expectedBoundaries, biddingZoneA);
+    }
+
+    @Test
+    void removeAreaBoundaries() {
+        Terminal boundary1 = network.getLine(EurostagTutorialExample1Factory.NHV1_NHV2_1).getTerminal1();
+        Terminal boundary2 = network.getGenerator("GEN").getTerminal();
+        Boundary boundary3 = network.getDanglingLine("danglingLine1").getBoundary();
+
+        aicA.removeAreaBoundary(boundary1);
+        Map<Object, Boolean> expectedBoundaries = Map.of(boundary2, true, boundary3, false);
+        assertBoundaries(expectedBoundaries, aicA);
+
+        aicA.removeAreaBoundary(boundary3);
+        expectedBoundaries = Map.of(boundary2, true);
+        assertBoundaries(expectedBoundaries, aicA);
+    }
+
+    @Test
+    void removeEquipmentRemovesAreaBoundary() {
+        assertEquals(3, aicA.getAreaBoundaryStream().count());
+
+        // Deleting equipment from the network should automatically delete their AreaBoundary-s
+        network.getLine(EurostagTutorialExample1Factory.NHV1_NHV2_1).remove();
+        network.getGenerator("GEN").remove();
+        network.getDanglingLine("danglingLine1").remove();
+
+        assertEquals(0, aicA.getAreaBoundaryStream().count());
+        assertBoundaries(Map.of(), aicA);
+    }
+
+    void assertBoundaries(Map<Object, Boolean> expectedBoundaries, Area area) {
```

Comment:
I don't know if this method, relying on objects equalities, will work with every IIDM implementations. It may be useful to have it in public.

Reply:
The method has been updated to check ids (and sides for terminals) rather that object equalities.
see 2796ed468ad8bb5c9ab62e89f3a2163925888bab
I let it private, but maybe we would still like to make it public ?

Reply:
Thanks for having rewritten this method.
I think you can leave it with this visibility. The checks are only done using String and enum values so it shouldn't depend on the implementation.  :+1: